rcl/test/rcl/test_namespace.cpp

```diff
@@ -0,0 +1,137 @@
+// Copyright 2016 Open Source Robotics Foundation, Inc.
```

Comment:
Also, please consider putting this file in `rcl/test` rather than `rcl/test/rcl`. I've tried to keep tests in the `rcl/test/rcl` folder a one-to-one match with source files in `rcl/src/rcl`. Since this doesn't correspond to a `rcl/src/rcl/namespace.c` file (AFAICT), I think it can be raised up a folder level.